sugar.go

```diff
@@ -112,31 +112,33 @@ func getSugarFields(args ...interface{}) ([]Field, error) {
 				return nil, errors.New("field name must be string")
 			}
 		} else {
-			switch value := value.(type) {
+			switch v := value.(type) {
+			case error:
+				return nil, errors.New("error can only be the first argument")
 			case bool:
-				fields = append(fields, Bool(key, value))
+				fields = append(fields, Bool(key, v))
 			case float64:
-				fields = append(fields, Float64(key, value))
+				fields = append(fields, Float64(key, v))
 			case int:
-				fields = append(fields, Int(key, value))
+				fields = append(fields, Int(key, v))
 			case int64:
-				fields = append(fields, Int64(key, value))
+				fields = append(fields, Int64(key, v))
 			case uint:
-				fields = append(fields, Uint(key, value))
+				fields = append(fields, Uint(key, v))
 			case uint64:
-				fields = append(fields, Uint64(key, value))
+				fields = append(fields, Uint64(key, v))
 			case string:
-				fields = append(fields, String(key, value))
+				fields = append(fields, String(key, v))
 			case time.Time:
-				fields = append(fields, Time(key, value))
+				fields = append(fields, Time(key, v))
 			case time.Duration:
-				fields = append(fields, Duration(key, value))
+				fields = append(fields, Duration(key, v))
 			case fmt.Stringer:
-				fields = append(fields, Stringer(key, value))
-			case error:
-				return nil, errors.New("error can only be the first argument")
+				fields = append(fields, Stringer(key, v))
+			case LogMarshaler:
```

Comment:
@akabos I bet (Just guessing) the LogMarshaler didn't understand unexported struct. I commented on the test below to update m9e to json serializable struct that seemed to fix the test coverage issue you are seeing.